carbons/carbons.go

```diff
@@ -63,3 +66,45 @@ func DisableIQ(ctx context.Context, s *xmpp.Session, iq stanza.IQ) error {
 	)), &v)
 	return err
 }
+
+type Received struct {
+	XMLName xml.Name `xml:"urn:xmpp:carbons:2 received"`
+}
+
+func (received Received) Wrap(r xml.TokenReader) xml.TokenReader {
+	return xmlstream.Wrap(
+		r,
+		xml.StartElement{Name: xml.Name{Space: NS, Local: "received"}},
+	)
+}
+
+func WrapReceived(received time.Time, r xml.TokenReader) xml.TokenReader {
+	return Received{}.Wrap(
+		forward.Forwarded{
+			Delay: delay.Delay{
+				Time: received,
+			},
+		}.Wrap(r),
+	)
+}
+
+type Sent struct {
```

Comment:
Same as above, let's just go with the functions for now. We can add structs later if we need them so let's drop this struct and its method and just make it part of `WrapSent`.